### ISSUE TITLE
Events afterRequest / afterOnLoad not fired in some cases

### ISSUE THREAD

**Issue by Ramblurr** [bigskysoftware/htmx#2547]
### Problem

The `htmx:afterRequest` and `htmx:afterOnLoad` events are not being fired when the DOM no longer contains the element nor any of its hierarchy.

### Background

I have a popup action menu powered by [tippy](https://atomiks.github.io/tippyjs/v6/getting-started/) that dynamically positions some buttons. These buttons cause htmx requests.

![2024-05-15_09-13-52](https://github.com/bigskysoftware/htmx/assets/14830/01a64032-96fa-4f54-b2a3-9b442dca27b7)

By the time the HTMX request is complete, the entire dom subtree of that menu (including the trigger button) have been removed from the DOM.

This results in `htmx:afterRequest` and `htmx:afterOnLoad` events  not being triggered.

### Debugging

![image](https://github.com/bigskysoftware/htmx/assets/14830/ec2cce0c-8730-4640-a727-e60411998e2a)

Indeed, the trigger's parent hierarchy bottoms (tops?) out without getting to `<body>`.
![2024-05-15_09-11-32](https://github.com/bigskysoftware/htmx/assets/14830/ab337664-b746-448e-834d-4efa868fdb4d)

**Comment by LapinskasL** [comment]
I'm encountering the same issue. Now it's making a bit more sense since it did not occur to me that it's not firing because the element is being removed.

**Comment by Ramblurr** [comment]
> Did you ever end up finding a solution?

No, just a workaround. I changed it so the popup doesn't disappear until the request is complete. In the meantime while the request is running, I show a spinner at the popup item that was clicked.

**Comment by Ramblurr** [comment]
I understand *why* this is the case currently, but it seems to me that the htmx:afterRequest / htmx:afterOnLoad should still be fired *somehow* even when the original element is gone.
